test/FunctionalTests/Infrastructure/GrpcTestFixture.cs

```diff
@@ -0,0 +1,63 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Net.Http;
+using FunctionalTestsWebsite.Infrastructure;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Grpc.AspNetCore.FunctionalTests.Infrastructure
+{
+    public class GrpcTestFixture<TStartup> : IDisposable where TStartup : class
+    {
+        private readonly TestServer _server;
+
+        public GrpcTestFixture()
+        {
+            Signal = new Signaler();
+
+            Action<IServiceCollection> configureServices = services =>
+            {
+                // Register signaler so server can signal tests
+                services.AddSingleton(Signal);
+            };
+
+            var builder = new WebHostBuilder()
+                .ConfigureServices(configureServices)
+                .UseStartup<TStartup>();
+
+            _server = new TestServer(builder);
+
+            Client = _server.CreateClient();
```

Comment:
qq: is this creating an actual server and an actual client?   Btw, I thought there is no currently HttpClient that'd support HTTP/2.